refl1d/probe.py

```diff
@@ -1725,36 +1764,44 @@ def shared_beam(self, intensity=1, background=0,
                 x.sample_broadening = sample_broadening
 
     def oversample(self, n=6, seed=1):
+        self._oversample(n, seed)
+        self.oversampling = n
+        self.oversampling_seed = seed
+
+    def _oversample(self, n=6, seed=1):
         # doc string is inherited from parent (see below)
         rng = numpy.random.RandomState(seed=seed)
         T = rng.normal(self.T[:, None], self.dT[:, None], size=(len(self.dT), n))
         L = rng.normal(self.L[:, None], self.dL[:, None], size=(len(self.dL), n))
         T = T.flatten()
         L = L.flatten()
         self._set_calc(T, L)
-    oversample.__doc__ = Probe.oversample.__doc__
+    _oversample.__doc__ = Probe.oversample.__doc__
 
     def _calculate_union(self):
-        theta_offsets = [x.theta_offset.value for x in self.xs if x is not None]
-        if self._theta_offsets is not None and theta_offsets == self._theta_offsets:
-            # no change in offsets: use cached values of measurement union
-            return
-
-        unique_offsets = set(theta_offsets)
-        shared_offset = theta_offsets[0] if len(unique_offsets) == 1 else None
-        if self._theta_offsets is not None \
-            and shared_offset is not None \
-            and len(set(self._theta_offsets)) == 1:
-            # offset was shared, and is shared, but value changed
-            Q = TL2Q(T=self.T + shared_offset, L=self.L)
-            self.calc_Qo = Q
+        theta_offset_limits = []
+        for xs in self.xs:
+            if xs is None or xs.theta_offset.fixed:
+                theta_offset_limits.append(None)
+            else:
+                theta_offset_limits.append(xs.theta_offset.bounds.limits)
+
+        if self._theta_offset_limits is not None and theta_offset_limits == self._theta_offset_limits:
+           # no change in offset limits: use cached values of measurement union
```

Comment:
The current code uses `theta_offset` value when calculating measurement union, so you would need to check for that changing as well.

Reply:
My logic was that if it was fixed, the value wouldn't be changing, so that you don't need to explicitly check the value.  Is there a situation where this will fail?

Reply:
Yes, it seems that the code should work like you expect. I just saw that `theta_offset.value` was being referenced; I didn't actually look at how it was being used.

Reply:
Okay, here's a scenario: In the GUI, fix theta offset range and change its value manually. It is still fixed, so no triggering of the recalc, but the Q_calc list will be different when you save and reload.

Reply:
Good catch!